solc/CommandLineInterface.cpp

```diff
@@ -563,7 +563,7 @@ bool CommandLineInterface::readInputFilesAndConfigureRemappings()
 				addStdin = true;
 			else
 			{
-				auto infile = boost::filesystem::path(path);
+				auto const infile = m_basePath / path; //boost::filesystem::path(path);
```

Comment:
This changes behaviour: if `--base-path` is specified, that will be prepended to the filename, e.g. there is no need to use a complete path.

Because of this it may be a breaking change.

Reply:
This sounds a bit weird to me - arguments to shell commands should always be files that can be resolved in the usual  way.

Shouldn't we rather do something like:

```
if (absolute_path(m_basePath).isPrefixOf(absolute_path(path)))
  path = absolute_path(path).removePrefix(absolute_path(m_basePath))
infile = m_basePath / path;
```
?

Reply:
I guess that can be a convenient feature, however couldn't find similar functions in boost to your pseudocode. Do you have any pointers?

Reply:
It was just pseudo code.